### ISSUE TITLE
Mapping inherited class properties

### ISSUE THREAD

**Issue by shiomax** [DapperLib/DapperAOT#93]
Right now when using a base class for your entities like this

```
public abstract class EntityBase
{
    public long Id { get; set; }
}

public class Entity1 : EntityBase
{
    public string Name { get; set; }
}
```

Then the ```Id``` property will not be read by the code generated.

It only works when putting the Id directly into ```Entity1``` (otherwise it never gets mapped, always default 0 for any query). I think it´s fairly common to want to enforce consistent naming of ids and timestamps. Although, this can be enforced with interfaces too I suppose (doing that now instead).

The above should either work, or a the analyzer should give you a warning about this (in case this is not gonna be implemented anytime soon). I don´t yet know enough about compile time generators to know how hard it is to figure out that there is a base class and walk back up the class tree.

**Comment by mgravell** [comment]
agreed, this should work; will get to it

**Comment by JerryJian** [comment]
Hello, when will this fix be released, thanks?

**Comment by Dean-NC** [comment]
This still doesn't work for me using the latest main branch as of today. Could you re-open this?

```
public abstract class DataModelBase
{
    public int DataLoadId { get; set; }
    public int LineNum { get; set; }
}

public sealed class TestModel : DataModelBase
{
    public string FirstName { get; set; }
    public string LastName { get; set; }
}
```
Below is from the TypeAccessor generated code:

![Image](https://github.com/user-attachments/assets/8e81918e-3f37-4d48-abcf-22dae998f948)

**Comment by Dean-NC** [comment]
I attached the VS debugger, and wherever TypeSymbol.GetMembers() is called, including in the Roslyn code itself, it doesn't see the inherited properties.

**Comment by Dean-NC** [comment]
Thanks for reopening this. I've checked again with version 1.0.48, but it's the same problem and result. I also tested with the base class being regular and not abstract, but same result. It seems this is a TypeAccessor generation issue.

**Comment by andreasblueher** [comment]
@mgravell 

I also looked into this issue. The first fix didn't work because TypeAccessorInterceptorGenerator.ConstructTypeMembers at line 539 of TypeAccessorInterceptorGenerator.cs still used a single typeSymbol.GetMembers() call without walking the hierarchy.

I've created https://github.com/andreasblueher/DapperAOT/tree/issue93 containing a fix for it. If you give me a sign, i could create a pull request
